packages/mdc-segmented-button/segmented-button/test/foundation.test.ts

```diff
@@ -0,0 +1,165 @@
+  };
+
+  const setupMultiSelectTest = () => {
+    const {foundation, mockAdapter} = setUpFoundationTest(MDCSegmentedButtonFoundation);
+    mockAdapter.hasClass.withArgs('mdc-segmented-button--single-select').and.returnValue(false);
+    return {foundation, mockAdapter};
+  };
+
+  /*** #isSingleSelect ***/
+
+  it('if single select, #isSingleSelect returns true', () => {
+    const {foundation} = setupSingleSelectTest()
+    expect(foundation.isSingleSelect()).toBeTruthy();
+  });
+
+  it('if multi select, #isSingleSelect returns false', () => {
+    const {foundation} = setupMultiSelectTest();
+    expect(foundation.isSingleSelect()).toBeFalsy();
+  });
+
+
+  /*** #isSegmentSelected ***/
+
+  it('if segment at index is selected, #isSegmenteSelected returns true', () => {
+
+  });
+
+  it('if segment with segmentId is selected, #isSegmenteSelected returns true', () => {
+
+  });
+
+  it('if segment at index is not selected, #isSegmenteSelected returns false', () => {
+
+  });
+
+  it('if segment with segmentId is not selected, #isSegmenteSelected returns false', () => {
+
+  });
+
+  it('if no segment at index, #isSegmenteSelected returns false', () => {
+
+  });
+
+  it('if no segment with segmentId, #isSegmenteSelected returns false', () => {
+
+  });
+
+  /*** getSelectedSegments ***/
+
+  it('#getSelectedSegments returns selected segments', () => {
+
+  });
+
+  /*** #selectSegment ***/
+
+  it('#selectSegment selects segment at index if it is unselected', () => {
+
+  });
+
+  it('#selectSegment selects segment with segmentId if it is unselected', () => {
+
+  });
+
+  it('#selectSegment does nothing if segment is already selected', () => {
+
+  });
+
+  //?
+  it('#selectSegment does nothing if segment does not exist', () => {
+
+  });
+
+  it('#selectSegment does not emit an event', () => {
+
+  });
+
+  /*** #unselectSegment ***/
+
+  it('#unselectSegment unselects segment at index if it is selected', () => {
+
+  });
+
+  it('#unselectSegment unselects segment with segmentId if it is selected', () => {
+
+  });
+
+  it('#unselectSegment does nothing if segment is already unselected', () => {
+
+  });
+
+  //?
+  it('#unselectSegment does nothing if segment does not exist', () => {
+
+  });
+
+  it('#unselectSegment does not emit an event', () => {
+
+  });
+
+  /*** #handleSelected ****/
+
+  // TODO: include this in every test
+  it('#handleSelected emits event with appropriate details', () => {
+
+  });
+
+  it('if single select, #handleSelected unselects segment if another segment was selected', () => {
+
+  });
+
+  it('if single select, #handleSelected reselects segment if it was unselected', () => {
+
+  });
+
+  it('if multi select, #handleSelected changes nothing if segment is selected', () => {
+
+  });
+
+  //? tests ability to unselect everything...this is not an obscure nor problematic edge case
+  it('if multi select, #handleSelected changes nothing if segment is unselected', () => {
```

Comment:
Are these tests too specific? I don't want to miss any test cases but am worried that I'm including unnecessary tests.